backend/src/modules/Boards/repository/mongo/MongoBoards.repository.js

```diff
@@ -30,6 +31,15 @@ export class MongoBoardsRepository {
     );
     return mongoDocumentToDomain(updatedBoard);
   }
+
+  async getBoardAnnouncements(boardID) {
+    const returnedData = await MongoBoardsModel.findById(boardID).select('announcements').exec();
+    const foundBoardAnnouncements = returnedData?.announcements?.map((announcement) =>
+      announcement.valueOf(),
+    );
+    console.log(foundBoardAnnouncements);
+    return foundBoardAnnouncements;
```

Comment:
```suggestion
    return mongoDocumentToDomain(foundBoardAnnouncements);
```
To powinno rozwiązać problem z:
 `if (!mongoose.isValidObjectId(boardId) && this.repository.constructor.name == 'MONGO') {`
 A przynajmniej mocno pomóc

Reply:
Chyba nie, bo tu zwracam po prostu tablicę announcementów, a mongoDocumentToDomain tworzy mi obiekt new Board

Reply:
![image](https://user-images.githubusercontent.com/41823689/158709744-0426c55c-07fa-496b-8179-6217927e499b.png)